metrics/slow_query.go

```diff
@@ -4,19 +4,20 @@ import (
 	"database/sql"
 	"errors"
 	"sync"
+	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 type SlowQueryMetrics struct {
-	mutex   sync.Mutex
-	metrics map[string]prometheus.Gauge
-	seconds int
+	mutex        sync.Mutex
+	metrics      map[string]prometheus.Gauge
+	milliseconds int64
 }
 
-func NewSlowQueryMetrics(secondsToConsiderSlow int) *SlowQueryMetrics {
+func NewSlowQueryMetrics(durationToConsiderSlow time.Duration) *SlowQueryMetrics {
 	return &SlowQueryMetrics{
-		seconds: secondsToConsiderSlow,
+		milliseconds: durationToConsiderSlow.Nanoseconds() / int64(time.Millisecond),
```

Comment:
Sorry that I wasn't clear enough. Idiomatic form is
```go
int64(durationToConsiderSlow / time.Millisecond)
```
This way you get a number of units on the right from any duration on the left.
And I'm sorry for nitpicking. :)

Reply:
https://golang.org/pkg/time/#Duration

Reply:
At least it's more beautiful.